src/components/data/apiContext.ts

```diff
@@ -13,14 +13,25 @@ type APIFunctions = typeof CommonAPI &
     typeof TaskAPI &
     typeof WorkflowAPI;
 
-export interface APIContextValue extends APIFunctions {}
+export interface LoginStatus {
+    expired: boolean;
+    setExpired(expired: boolean): void;
+}
+export interface APIContextValue extends APIFunctions {
+    loginStatus: LoginStatus;
+}
+
 export const defaultAPIContextValue = {
     ...CommonAPI,
     ...ExecutionAPI,
     ...LaunchAPI,
     ...ProjectAPI,
     ...TaskAPI,
-    ...WorkflowAPI
+    ...WorkflowAPI,
+    loginStatus: {
```

Comment:
Is this going to be incorporated into another API in the future? Feels somewhat odd to be the only item in the context not spread from some other imported value. Just a style thing, though.

Reply:
I probably should have documented this!
This won't be part of an API. This is an application state value that tells us if a request has failed with a 401. It's part of this context for two reasons:
1. Any component/hook making an API request can see and/or set the loginStatus. At the moment, this is only used through the `useFetchableData` base hook.
2. APIContext gets exposed by a provider at the root of the component tree, allowing the blanket "your session expired" component to consume it.